src/components/graph/search/GraphSearchDialog.tsx

```diff
@@ -281,15 +283,34 @@ const GraphSearchDialog = (props: Props) => {
 				.getMessage(id, abortController.signal, { probe: true })
 				.then(foundMessage => {
 					if (foundMessage !== null) {
-						onSearchSuccess(foundMessage, searchTimestamp);
+						foundMessage.parsedMessages?.forEach((parsedMessage, index) => {
```

Comment:
this code repeats at least 3 times, maybe we should extract it to a helper function?